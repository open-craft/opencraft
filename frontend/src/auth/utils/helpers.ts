import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
