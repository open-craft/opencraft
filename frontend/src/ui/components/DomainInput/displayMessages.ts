const messages = {
  typeDomainNameBelow: {
    defaultMessage: 'Type your domain name below',
    description: 'Message above domain input text box.'
  },
  checkAvailability: {
    defaultMessage: 'Check Availability',
    description: 'Label for button that check availability of Domain'
  }
};

export default messages;
