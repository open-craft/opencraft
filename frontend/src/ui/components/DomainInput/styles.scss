@import "~styles/theme";

.domain-input-container {
  width: 100%;

  .domain-label {
    font-family: Chivo;
    font-size: 18px;
    letter-spacing: 0.54px;
    color: #0f1f24;
    margin-bottom: 20px;
  }

  .input-group {
    font-family: $chivo-font;
    font-size: 16px;
    letter-spacing: 0.96px;
    color: #000000;

    .form-control {
      border: 1px solid $primary-1;
      border-right-width: 0px;
    }
    .input-group-text {
      height: 38px;
      border: 1px solid $primary-1;
      background-color: white;
      color: black;
    }

    button {
      height: 38px;
      width: 220px;
      vertical-align: middle;
      font-size: 13px;
      .spinner {
        margin-right: 10px;
      }
    }
  }
}
