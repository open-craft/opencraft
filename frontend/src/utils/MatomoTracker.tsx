import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
