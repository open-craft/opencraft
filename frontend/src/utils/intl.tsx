import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
