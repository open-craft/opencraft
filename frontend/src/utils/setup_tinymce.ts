import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
