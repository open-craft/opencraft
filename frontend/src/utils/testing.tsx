import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
