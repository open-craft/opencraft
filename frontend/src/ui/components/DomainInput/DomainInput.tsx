import * as React from 'react';
import { Button, FormControl, InputGroup, Spinner } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import { INTERNAL_DOMAIN_NAME } from 'global/constants';
import messages from './displayMessages';
import './styles.scss';

interface Props {
  domainName: string;
  internalDomain: boolean;
  loading: boolean;
  error: string;
  handleDomainChange: Function;
  handleSubmitDomain: Function;
}

export const DomainInput: React.SFC<Props> = (props: Props) => {
  return (
    <div className="domain-input-container">
      <div className="domain-label">
        <WrappedMessage messages={messages} id="typeDomainNameBelow" />
      </div>
      <InputGroup>
        <FormControl
          value={props.domainName}
          onChange={(event: any) => {
            props.handleDomainChange(event.target.value);
          }}
          isInvalid={!!props.error}
        />
        {props.internalDomain && (
          <InputGroup.Append>
            <InputGroup.Text>{INTERNAL_DOMAIN_NAME}</InputGroup.Text>
          </InputGroup.Append>
        )}
        <InputGroup.Append>
          <Button
            variant="primary"
            disabled={false || props.loading}
            onClick={(event: any) => {
              props.handleSubmitDomain();
            }}
          >
            {props.loading && (
              <Spinner animation="border" size="sm" className="spinner" />
            )}
            <WrappedMessage messages={messages} id="checkAvailability" />
          </Button>
        </InputGroup.Append>
        {props.error && (
          <FormControl.Feedback type="invalid">
            {props.error}
          </FormControl.Feedback>
        )}
      </InputGroup>
    </div>
  );
};
