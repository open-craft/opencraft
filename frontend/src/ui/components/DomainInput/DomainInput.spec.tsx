import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { DomainInput } from './DomainInput';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<DomainInput />).toJSON();
    expect(tree).toMatchSnapshot();
});
