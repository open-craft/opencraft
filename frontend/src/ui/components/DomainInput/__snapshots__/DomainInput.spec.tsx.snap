// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="domain-input-container"
>
  <div
    className="domain-label"
  >
    Type your domain name below
  </div>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Check Availability
      </button>
    </div>
  </div>
</div>
`;
